File: Bin/Data/LuaScripts/Utilities/Network.lua
runServer = false
runClient = false
serverAddress = ''
serverPort = 1234
userName = ''
nobgm = false

function ParseNetworkArguments()
    local arguments = GetArguments()
    local skipNext = false
    for i, argument in ipairs(arguments) do
        if skipNext == false and string.sub(argument, 1, 1) == '-'  then
            argument = string.lower(string.sub(argument, 1))
            if argument == "server" then
                runServer = true
                runClient = false
            elseif argument == "address" then
                runClient = true
                runServer = false
                serverAddress = arguments[i + 1]
                skipNext = true
            elseif argument == "username" then
                userName = arguments[i + 1]
                skipNext = true
            elseif argument == "nobgm" then
                nobgm = true
            end
        end
        skipNext = false
    end
end

File: Source/Samples/ninja_snow_war/utilities/network.cpp
// Copyright (c) 2022-2023 the Dviglo project
// Copyright (c) 2008-2023 the Urho3D project
// License: MIT

#include <Urho3D/Core/ProcessUtils.h> // GetArguments()

#include "network.h"

namespace Urho3D
{

void ParseNetworkArguments()
{
    const Vector<String>& arguments = GetArguments();

    for (i32 i = 0; i < arguments.Size(); ++i)
    {
        String argument = arguments[i].ToLower();

        if (argument[0] == '-')
        {
            argument = argument.Substring(1);

            if (argument == "server")
            {
                runServer = true;
                runClient = false;
            }
            else if (argument == "address")
            {
                runClient = true;
                runServer = false;
                serverAddress = arguments[i + 1];
                ++i;
            }
            else if (argument == "username")
            {
                userName = arguments[i + 1];
                ++i;
            }
            else if (argument == "nobgm")
            {
                nobgm = true;
            }
        }
    }
}

} // namespace Urho3D
